src/renderer/widgets/AddProxyModal/ui/SubmitProxy.tsx

```diff
@@ -0,0 +1,56 @@
+import { useUnit } from 'effector-react';
+import { ComponentProps, useEffect } from 'react';
+
+import { Button } from '@shared/ui';
+import { useI18n } from '@app/providers';
+import { useTaskQueue } from '@shared/lib/hooks';
+import { OperationResult } from '@entities/transaction';
+import { useMultisigTx, useMultisigEvent } from '@entities/multisig';
+import { SubmitStep } from '../lib/types';
+import { submitProxyUtils } from '../lib/submit-proxy-utils';
+import { submitModel } from '../model/submit-model';
+
+type ResultProps = Pick<ComponentProps<typeof OperationResult>, 'title' | 'description' | 'variant'>;
+
+type Props = {
+  isOpen: boolean;
+  onClose: () => void;
+};
+
+export const SubmitProxy = ({ isOpen, onClose }: Props) => {
+  const { t } = useI18n();
+
+  const submitStore = useUnit(submitModel.$submitStore);
+  const { step, message } = useUnit(submitModel.$submitStep);
+
+  const { addTask } = useTaskQueue();
+  const { addMultisigTx } = useMultisigTx({ addTask });
+  const { addEventWithQueue } = useMultisigEvent({ addTask });
+
+  useEffect(() => {
+    submitModel.events.hooksApiChanged({ addMultisigTx, addEventWithQueue });
+  }, [addMultisigTx, addEventWithQueue]);
```

Comment:
Am I right that this is the same as `gate` or is there some difference?

Reply:
I assume it's the same as Gate (not a big fan of it, another react specific function)
Event style is just more generic
Also we used to do `apiChanged` in other components, good to be consistent